editor/components/block-list/style.scss

```diff
@@ -61,48 +42,81 @@
 		background-color: rgba( $white, 0.6 );
 	}
 
-	// simpler style for a block that has cursor focus (but hasn't been selected)
-	&.is-selected > .editor-block-mover:before,
-	&.is-hovered > .editor-block-mover:before,
-	&.is-selected > .editor-block-settings-menu:before,
-	&.is-hovered > .editor-block-settings-menu:before {
+
+	/**
+	 * Hovered Block style
+	 */
+
+	&.is-selected .editor-block-mover:before,
+	&.is-hovered .editor-block-mover:before,
+	&.is-selected .editor-block-settings-menu:before,
+	&.is-hovered .editor-block-settings-menu:before {
 		content: '';
 		position: absolute;
 		height: 36px;
 		top: 10px;
 	}
 
-	&.is-selected > .editor-block-mover:before,
-	&.is-hovered > .editor-block-mover:before {
+	&.is-selected .editor-block-mover:before,
+	&.is-hovered .editor-block-mover:before {
 		border-right: 1px solid $light-gray-500;
 		right: 6px;
 	}
 
-	&.is-reusable.is-selected > .editor-block-mover:before {
+	&.is-reusable.is-selected .editor-block-mover:before {
 		border-right: none;
 	}
 
-	&.is-selected > .editor-block-settings-menu:before,
-	&.is-hovered > .editor-block-settings-menu:before {
+	&.is-selected .editor-block-settings-menu:before,
+	&.is-hovered .editor-block-settings-menu:before {
 		border-left: 1px solid $light-gray-500;
 		left: 6px;
 	}
 
-	&.is-reusable.is-selected > .editor-block-settings-menu:before {
+	&.is-reusable.is-selected .editor-block-settings-menu:before {
 		border-left: none;
 	}
 
+
+	/**
+	 * Selected Block style
+	 */
+
+	.editor-block-list__block-edit>div:not( .components-drop-zone ):not( .editor-block-mover ):not( .editor-block-settings-menu ):not( .editor-block-contextual-toolbar ):not( .editor-block-list__block-mobile-toolbar ) {	// @todo — target a classname, not this crazy "not" rule
+		position: relative;
+
+		&:before {
+			z-index: z-index( '.editor-block-list__block:before' );
+			content: '';
+			position: absolute;
+			top: -$block-padding;
```

Comment:
It's quite nice to be able to use negative values here. That means we paint the outline around the content itself, allowing the content margins to position and space items. The border is just painted around that, as is also shown in https://codepen.io/joen/pen/aqKLqa. 

This should benefit nested UI also, as it will be one less bit of UI to "offset" in nested contexts.